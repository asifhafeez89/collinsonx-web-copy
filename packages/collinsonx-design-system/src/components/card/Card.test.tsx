import * as React from 'react';
import renderer from 'react-test-renderer';
import Card from '.';

const weekdays: string = 'Monday, Tuesday';
const price = {
  currency: 'USD',
  reservationCost: 17.5,
  lifestyleXReservationCharge: 20.5,
};

describe('Card component', () => {
  it('renders Container', () => {
    const tree = renderer
      .create(
        <Card
          handleClick={() => console.log()}
          title="My lounge"
          subtitle="Athens International Airport"
          price={price}
          openingHours={weekdays}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
