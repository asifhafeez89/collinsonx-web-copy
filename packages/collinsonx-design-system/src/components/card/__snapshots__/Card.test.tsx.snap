// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card component renders Container 1`] = `
.emotion-0 {
  width: 100%;
  -webkit-transition: 0.3s;
  transition: 0.3s;
  border-radius: 5px;
  background-color: #fff;
  padding: 1rem;
  margin-bottom: 1rem;
  box-shadow: 0px 6px 15px rgba(0, 0, 0, 0.15);
}

.emotion-0 h3 {
  padding-bottom: 0px;
  margin-bottom: 0px;
}

.emotion-0 p {
  margin-top: 0px;
}

.emotion-0:hover {
  box-shadow: 10 8px 16px 10 rgba(0, 0, 0, 0.2);
}

.emotion-0 .currency:not(:empty):before {
  content: 'Â£';
  font-size: 14px;
  vertical-align: 10px;
}

.emotion-1 {
  padding-bottom: 8px;
}

.emotion-1 h3 {
  margin-top: 8px;
}

.emotion-3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  -webkit-box-pack: start;
  -ms-flex-pack: start;
  -webkit-justify-content: flex-start;
  justify-content: flex-start;
  gap: 0.5rem;
  padding-bottom: 1rem;
}

.emotion-4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 0.625rem;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-5 {
  min-width: 1rem;
}

.emotion-7 {
  font-family: -apple-system,BlinkMacSystemFont,Segoe UI,Roboto,Helvetica,Arial,sans-serif,Apple Color Emoji,Segoe UI Emoji;
  -webkit-tap-highlight-color: transparent;
  color: inherit;
  font-size: inherit;
  line-height: 1.55;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-weight: 600;
}

.emotion-7:focus {
  outline-offset: 0.125rem;
  outline: 0.125rem solid #339af0;
}

.emotion-7:focus:not(:focus-visible) {
  outline: none;
}

.emotion-11 {
  font-family: -apple-system,BlinkMacSystemFont,Segoe UI,Roboto,Helvetica,Arial,sans-serif,Apple Color Emoji,Segoe UI Emoji;
  -webkit-tap-highlight-color: transparent;
  color: inherit;
  font-size: inherit;
  line-height: 1.55;
  -webkit-text-decoration: none;
  text-decoration: none;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
  font-weight: 600;
}

.emotion-11:focus {
  outline-offset: 0.125rem;
  outline: 0.125rem solid #339af0;
}

.emotion-11:focus:not(:focus-visible) {
  outline: none;
}

.emotion-13 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  -webkit-box-pack: start;
  -ms-flex-pack: start;
  -webkit-justify-content: flex-start;
  justify-content: flex-start;
  gap: 1rem;
}

.emotion-15 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 0.5rem;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
}

.emotion-18 {
  -webkit-tap-highlight-color: transparent;
  font-family: -apple-system,BlinkMacSystemFont,Segoe UI,Roboto,Helvetica,Arial,sans-serif,Apple Color Emoji,Segoe UI Emoji;
  cursor: pointer;
  border: 0;
  padding: 0;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  font-size: 1rem;
  background-color: transparent;
  text-align: left;
  color: #000;
  -webkit-text-decoration: none;
  text-decoration: none;
  box-sizing: border-box;
  height: 2.25rem;
  padding-left: 1.125rem;
  padding-right: 1.125rem;
  font-family: -apple-system,BlinkMacSystemFont,Segoe UI,Roboto,Helvetica,Arial,sans-serif,Apple Color Emoji,Segoe UI Emoji;
  -webkit-tap-highlight-color: transparent;
  display: block;
  width: 100%;
  border-radius: 0.25rem;
  font-weight: 600;
  position: relative;
  line-height: 1;
  font-size: 0.875rem;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  cursor: pointer;
  border: 0.0625rem solid #228be6;
  background-color: transparent;
  color: #228be6;
}

.emotion-18:focus {
  outline-offset: 0.125rem;
  outline: 0.125rem solid #339af0;
}

.emotion-18:focus:not(:focus-visible) {
  outline: none;
}

.emotion-18:focus {
  outline-offset: 0.125rem;
  outline: 0.125rem solid #339af0;
}

.emotion-18:focus:not(:focus-visible) {
  outline: none;
}

@media (hover: hover) {
  .emotion-18:hover {
    background-color: rgba(231, 245, 255, 0.35);
  }
}

@media (hover: none) {
  .emotion-18:active {
    background-color: rgba(231, 245, 255, 0.35);
  }
}

.emotion-18:active {
  -webkit-transform: translateY(0.0625rem);
  -moz-transform: translateY(0.0625rem);
  -ms-transform: translateY(0.0625rem);
  transform: translateY(0.0625rem);
}

.emotion-18:disabled,
.emotion-18[data-disabled] {
  border-color: transparent;
  background-color: #e9ecef;
  color: #adb5bd;
  cursor: not-allowed;
  background-image: none;
  pointer-events: none;
}

.emotion-18:disabled:active,
.emotion-18[data-disabled]:active {
  -webkit-transform: none;
  -moz-transform: none;
  -ms-transform: none;
  transform: none;
}

.emotion-18[data-loading] {
  pointer-events: none;
}

.emotion-18[data-loading]::before {
  content: "";
  position: absolute;
  top: -0.0625rem;
  right: -0.0625rem;
  left: -0.0625rem;
  bottom: -0.0625rem;
  background-color: rgba(255, 255, 255, .5);
  border-radius: 0.25rem;
  cursor: not-allowed;
}

.emotion-19 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  height: 100%;
  overflow: visible;
}

.emotion-21 {
  white-space: nowrap;
  height: 100%;
  overflow: hidden;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-25 {
  -webkit-tap-highlight-color: transparent;
  font-family: -apple-system,BlinkMacSystemFont,Segoe UI,Roboto,Helvetica,Arial,sans-serif,Apple Color Emoji,Segoe UI Emoji;
  cursor: pointer;
  border: 0;
  padding: 0;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  font-size: 1rem;
  background-color: transparent;
  text-align: left;
  color: #000;
  -webkit-text-decoration: none;
  text-decoration: none;
  box-sizing: border-box;
  height: 2.25rem;
  padding-left: 1.125rem;
  padding-right: 1.125rem;
  font-family: -apple-system,BlinkMacSystemFont,Segoe UI,Roboto,Helvetica,Arial,sans-serif,Apple Color Emoji,Segoe UI Emoji;
  -webkit-tap-highlight-color: transparent;
  display: block;
  width: 100%;
  border-radius: 0.25rem;
  font-weight: 600;
  position: relative;
  line-height: 1;
  font-size: 0.875rem;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  cursor: pointer;
  border: 0.0625rem solid transparent;
  background-color: #228be6;
  color: #fff;
}

.emotion-25:focus {
  outline-offset: 0.125rem;
  outline: 0.125rem solid #339af0;
}

.emotion-25:focus:not(:focus-visible) {
  outline: none;
}

.emotion-25:focus {
  outline-offset: 0.125rem;
  outline: 0.125rem solid #339af0;
}

.emotion-25:focus:not(:focus-visible) {
  outline: none;
}

@media (hover: hover) {
  .emotion-25:hover {
    background-color: #1c7ed6;
  }
}

@media (hover: none) {
  .emotion-25:active {
    background-color: #1c7ed6;
  }
}

.emotion-25:active {
  -webkit-transform: translateY(0.0625rem);
  -moz-transform: translateY(0.0625rem);
  -ms-transform: translateY(0.0625rem);
  transform: translateY(0.0625rem);
}

.emotion-25:disabled,
.emotion-25[data-disabled] {
  border-color: transparent;
  background-color: #e9ecef;
  color: #adb5bd;
  cursor: not-allowed;
  background-image: none;
  pointer-events: none;
}

.emotion-25:disabled:active,
.emotion-25[data-disabled]:active {
  -webkit-transform: none;
  -moz-transform: none;
  -ms-transform: none;
  transform: none;
}

.emotion-25[data-loading] {
  pointer-events: none;
}

.emotion-25[data-loading]::before {
  content: "";
  position: absolute;
  top: -0.0625rem;
  right: -0.0625rem;
  left: -0.0625rem;
  bottom: -0.0625rem;
  background-color: rgba(255, 255, 255, .5);
  border-radius: 0.25rem;
  cursor: not-allowed;
}

<div
  className="emotion-0"
>
  <div
    className="emotion-1"
  >
    <h3>
      My lounge
    </h3>
  </div>
  <div
    className="emotion-2 emotion-3"
  >
    <div
      className="emotion-4"
    >
      <div
        className="emotion-5"
      >
        <test-file-stub
          color="#0C8599"
          width={16}
        />
      </div>
      <div
        className="emotion-6 emotion-7"
      >
        Athens International Airport
      </div>
    </div>
    <div
      className="emotion-4"
    >
      <div
        className="emotion-5"
      >
        <test-file-stub
          color="#0C8599"
          width={16}
        />
      </div>
      <div
        className="emotion-6 emotion-11"
      >
        Monday, Tuesday
      </div>
    </div>
  </div>
  <div
    className="emotion-2 emotion-13"
    style={
      {
        "borderRadius": "5px 5px 0 0",
        "borderTop": "1px solid #c8c9ca",
      }
    }
  >
    <div
      className="emotion-14"
    >
      <p
        className="currency"
        style={
          {
            "color": "#0C8599",
            "fontSize": "28px",
            "fontWeight": "700",
            "margin": "0",
            "marginBottom": "4px",
          }
        }
      >
        17.50
      </p>
    </div>
    <div
      className="emotion-15"
    >
      <button
        className="emotion-16 emotion-17 emotion-18"
        data-button={true}
        style={
          {
            "border": "2px solid black",
            "color": "black",
          }
        }
        type="button"
      >
        <div
          className="emotion-19 emotion-20"
        >
          <span
            className="emotion-21 emotion-22"
            style={
              {
                "textTransform": undefined,
              }
            }
          >
            Scan QR
          </span>
        </div>
      </button>
      <button
        className="emotion-16 emotion-17 emotion-25"
        data-button={true}
        onClick={[Function]}
        type="button"
      >
        <div
          className="emotion-19 emotion-20"
        >
          <span
            className="emotion-21 emotion-22"
            style={
              {
                "textTransform": undefined,
              }
            }
          >
            View Lounge
          </span>
        </div>
      </button>
    </div>
  </div>
</div>
`;
